windows-agent/internal/proservices/landscape/executor_test.go

```diff
@@ -148,47 +148,40 @@ func TestReceiveCommandStartStop(t *testing.T) {
-
-	emptyFileChecksum := "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
-	mockErrorChecksum := "afe55cda4210c2439b47c62c01039027522f7ed4abdb113972b3030b3359532a"
-	mockMismatchChecksum := "1234"
-
 	testCases := map[string]struct {
 		noCloudInit            bool
 		cloudInitWriteErr      bool
 		distroAlreadyInstalled bool
-		emptyDistroName        bool
+		distroName             interface{} // using interface{} to allow nil, modelling an optional value.
 		wslInstallErr          bool
 		appxDoesNotExist       bool
 		nonResponsiveServer    bool
 		breakVhdxDir           bool
 		breakTarFile           bool
 		breakTempDir           bool
 
-		sendRootfsURL      string
-		sendRootfsChecksum string
+		sendRootfsURL    string
+		missingChecksums bool
 
 		wantCouldInitWriteCalled bool
 		wantInstalled            bool
 	}{
 		"From the store":                    {wantInstalled: true, wantCouldInitWriteCalled: true},
-		"From a rootfs URL":                 {sendRootfsURL: "goodfile", wantInstalled: true},
-		"From a rootfs URL with a checksum": {sendRootfsURL: "goodfile", sendRootfsChecksum: emptyFileChecksum, wantInstalled: true},
+		"From a rootfs URL with a checksum": {sendRootfsURL: "goodfile", wantInstalled: true},
 		"With no cloud-init":                {noCloudInit: true, wantCouldInitWriteCalled: true, wantInstalled: true},
 
-		"Error when the distroname is empty":         {emptyDistroName: true},
 		"Error when the Appx does not exist":         {appxDoesNotExist: true},
 		"Error when the distro is already installed": {distroAlreadyInstalled: true, wantInstalled: true},
 		"Error when the distro fails to install":     {wslInstallErr: true},
 		"Error when cannot write cloud-init file":    {cloudInitWriteErr: true, wantCouldInitWriteCalled: true},
+		"Error when the distroname is empty":         {distroName: ""},
 
-		"Error when the rootfs isn't a valid tarball":                   {sendRootfsURL: "badfile", sendRootfsChecksum: mockErrorChecksum, wantInstalled: false},
-		"Error when the checksum doesn't match":                         {sendRootfsURL: "goodfile", sendRootfsChecksum: mockMismatchChecksum, wantInstalled: false},
+		"Error when the distro ID is reserved":                          {sendRootfsURL: "goodfile", distroName: "Ubuntu-19.13", wantInstalled: false},
+		"Error when the rootfs isn't a valid tarball":                   {sendRootfsURL: "badfile", wantInstalled: false},
+		"Error when the checksum file doesn't exist":                    {missingChecksums: true, sendRootfsURL: "goodfile", wantInstalled: false},
+		"Error when the checksum doesn't match":                         {sendRootfsURL: "badchecksum", wantInstalled: false},
 		"Error when the rootfs doesn't exist":                           {sendRootfsURL: "badresponse", wantInstalled: false},
+		"Error when the rootfs URL is ill-formed":                       {sendRootfsURL: "https://?", wantInstalled: false},
```

Comment:
I am surprise this didn’t trigger url.Parse() erroring on line 162 of `executor.go`. Maybe this is an ill-formed that fails `url.Get` but not the parsing part mind checking? :)

I understand that line 346 can’t be covered (as we will fail on the rootfs url before the sha256 sums one), but I think 162 should be doable.

Reply:
Indeed it didn't break the URL parsing. That's because the URL always started with a valid scheme and address (coming from the `fileServerAddr` variable (assigned from the listener address, so always valid). I had to make that variable hold invalid contents to make `url.Parse` really break..

Reply:
exactly! I’m resolving this one but adding a comment on another approach we should take to avoid having another parameter + having to use a parameter that is ignored then.